### PR TITLE
feat: combine auctionSurplus and transferSurplus in one method

### PR DESCRIPTION
**Pull request by cupOJoseph**
## Motivation
Selling system coin from surplus creates too much sell pressure and prevents the DAO/governor from accruing any fees in system coin. The stablecoin would be more stable if a percent of fees went to a treasury instead of being sold.

# changes
Changes are kept simple by not changing function names or removing any functions, even if it adds some redundancy. 

1. Either `transferExtraSurplus()` or `auctionSurplus()` can be used to kick off an auction AND transfer some surplus
2. `surplusIsTransferred` parameter is no longer boolean, but is a percentage from 0-100 of what percent of surplus is to be transfered to the `extraSurplusReceiver` vs auctioned. 
3. the same checks for if the system has enough surplus to send should always be made
4. notes in the params script files are updated

### CODE REVIEW

**Comment by cupOJoseph** [comment]
TODO: update tests with correct expected values

File: docs/src/detailed/modules/acc_engine.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ The Accounting Engine serves as the system's financial management hub, overseein
 - **Surplus Auction House**: Is called to start surplus auctions.
 - **Debt Auction House**: Is called to start debt auctions.
 - `postSettlementSurplusDrain`: Address to which surplus is sent following Global Settlement.
-- `surplusIsTransferred`: Whether the surplus should be either auctioned off or transferred.
+- `surplusIsTransferred`: Percent of the surplus that should be transfered vs auctioned off.
```
Comment (wei3erHase) [inline docs/src/detailed/modules/acc_engine.md:48]: i'd rename it to `surplusTransferPercentage`

File: src/contracts/AccountingEngine.sol [hunk 2]
```diff
@@ -193,7 +193,7 @@ contract AccountingEngine is Authorizable, Modifiable, Disableable, IAccountingE
 
   /// @inheritdoc IAccountingEngine
   function auctionSurplus() external returns (uint256 _id) {
-    if (_params.surplusIsTransferred == 1) revert AccEng_SurplusAuctionDisabled();
+    if (_params.surplusIsTransferred == 100) revert AccEng_SurplusAuctionDisabled();
```
Comment (wei3erHase) [inline src/contracts/AccountingEngine.sol:196]: this should be in `WAD %` units

File: src/contracts/AccountingEngine.sol [hunk 3]
```diff
@@ -205,17 +205,31 @@ contract AccountingEngine is Authorizable, Modifiable, Disableable, IAccountingE
...
+    emit AuctionSurplus(_id, 0, _params.surplusAmount * (100 - _params.surplusIsTransferred) / 100);
+
+    //Transfer remaining surplus percentage
+    if(_params.surplusIsTransferred > 0){
+      if (extraSurplusReceiver == address(0)) revert AccEng_NullSurplusReceiver();
+
+      safeEngine.transferInternalCoins({
+        _source: address(this),
+        _destination: extraSurplusReceiver,
+        _rad: _params.surplusAmount * _params.surplusIsTransferred / 100
```
[the remark this answers is not in the document]
Comment (wei3erHase) [reply src/contracts/AccountingEngine.sol:220]: units are ok, `RAD.wmul(WAD) = RAD` (bc `wmul(WAD)` cancels itself)

File: src/contracts/AccountingEngine.sol
Comment (wei3erHase) [inline src/contracts/AccountingEngine.sol:1]: blz also add a parameter validation to make sure that the percentage is lesser than WAD (cannot transfer 105%)

**Review by wei3erHase — COMMENTED** [review]
missing:
- merge the 2 methods: transfer and auction (i'd still call it `auctionSurplus`)
- rm the Job method that calls transferSurplus
- fix units
- add validation to the parameter (lesser than WAD)

**Comment by wei3erHase** [comment]
Closing in favour of https://github.com/defi-wonderland/hai-core/pull/33
Not expected to be merged before deployment because of being un-audited tho.

File: src/contracts/AccountingEngine.sol (hunk 2 above)
Comment (daopunk) [reply src/contracts/AccountingEngine.sol:196]: isn't `_params.surplusIsTransferred` in RAD?

**Pull request closed without merging**